### DISCUSSION TITLE
make new rule selected by default

### DISCUSSION THREAD

**Discussion by kuljeet2604** [react-querybuilder/react-querybuilder#658] [General]
i want my initial state of query to be first like we pressed +rule button

**Comment by jakeboone02** [comment]
Either of the following configurations will work:

```tsx
const defaultQuery = { combinator: 'and', rules: [
  { field: 'defaultField', operator: '=', value: 'default value' }
]};

const App = () => (
  <QueryBuilder defaultQuery={defaultQuery} />
);
```

```tsx
const App = () => (
  <QueryBuilder addRuleToNewGroups />
);
```

Docs for the second example here: https://react-querybuilder.js.org/docs/components/querybuilder#addruletonewgroups

Note that the second example only works if both `query` and `defaultQuery` props are undefined.

**Reply by chinku04** [reply]
my value editor retains the selected value when the operator changes, can we reset the field, once the operator changes?

**Reply by jakeboone02** [reply]
[`resetOnFieldChange`](https://react-querybuilder.js.org/docs/components/querybuilder#resetonfieldchange) will reset the operator and the value when the field changes. **This is the default.**

[`resetOnFieldChange={false}`](https://react-querybuilder.js.org/docs/components/querybuilder#resetonfieldchange) will _not_ update the operator or the value when the field changes.

[`resetOnOperatorChange`](https://react-querybuilder.js.org/docs/components/querybuilder#resetonoperatorchange) will reset the value when the operator changes.

[`resetOnOperatorChange={false}`](https://react-querybuilder.js.org/docs/components/querybuilder#resetonoperatorchange) will _not_ reset the value when the operator changes. **This is the default.**

**Reply by chinku04** [reply]
This works, but if i supply options parameters (for single_select or multi_select ) in my fields.

first options from the options list is selected by default.

reseting works fine with fields that does not require option prop

**Reply by jakeboone02** [reply]
This was addressed in #645, which will be included in v7. A beta release of v7 is available under the `next` label (`npm i react-querybuilder@next`).

**Comment by kuljeet2604** [comment]
i tried v7 still first value (if options value is supplied), gets selected for input type text.
Options are meant for Multi-select or select  operator type.

**Reply by jakeboone02** [reply]
Can you put together a sandbox the demonstrates that behavior? I can't find anywhere in the code where that would happen.

**Reply by kuljeet2604** [reply]
https://codesandbox.io/p/devbox/interesting-hamilton-swc7nv?file=%2Fsrc%2FApp.tsx%3A69%2C5-69%2C11

kindly ignore other props as it is for my custom value editor

**Reply by jakeboone02** [reply]
Ah, ok. This has to do with the `resetOnOperatorChange` prop. I'll fix it in #663.
